win32ss/reactx/ntddraw/intddraw.h

```diff
@@ -108,6 +108,11 @@ typedef VOID (APIENTRY *PGD_ENGFREEPRIVATEUSERMEM)(PDD_SURFACE_LOCAL, PVOID);
 typedef PDD_SURFACE_LOCAL (APIENTRY *PGD_ENGLOCKDIRECTDRAWSURFACE)(HANDLE);
 typedef BOOL (APIENTRY *PGD_ENGUNLOCKDIRECTDRAWSURFACE)(PDD_SURFACE_LOCAL);
 
+/* Other internal functions */
+typedef VOID (APIENTRY *PGD_DXDDDYNAMICMODECHANGE)(HDEV, HDEV, ULONG);
+typedef VOID (APIENTRY *PGD_DXDDRESUMEDIRECTDRAW)(HDEV, ULONG);
+typedef VOID (APIENTRY *PGD_DXDDSUSPENDDIRECTDRAW)(HDEV, ULONG);
```

Comment:
Question: is this last parameter (the ULONG) truly a ULONG, or ULONG_PTR (or PVOID) ? (i.e. something whose size depends on 32 vs. 64 bits)

Reply:
As I know, `ULONG_PTR` is like a pointer to `ULONG`, same as `PVOID` is a pointer to `VOID`. In debugger, when I analysed this stuff, I saw only `00000000`, or `00000002` passed as the 2nd parameter into `DxDdSuspendDirectDraw` and `DxDdResumeDirectDraw`, and the 3rd one to `DxDdDynamicModeChange`. If it would be a pointer, it could look like a some hexadecimal address (combination of symbols), e. g. `k3ud58jm`, not just a decimal number, like `0` or `2`. Such pointer needs to be unassembled one more time to see its actual data. So it's looks more like a simple `ULONG` or `DWORD` flags, those are not documented as well.

Reply:
`ULONG_PTR` is a scalar value (a number) whose size corresponds to the size of pointers in the current platform.
For example, a `ULONG_PTR` is 4 bytes when you compile for 32-bits , but is 8 bytes when you compile for 64-bits.
This is why such a value can hold pointer addresses without problems.

On the contrary, using `ULONG` values to convey pointer addresses, would truncate such addresses when compiling code for 64-bits.

The type for a pointer to `ULONG` , i.e. `ULONG *`, is named `PULONG` usually, and not `ULONG_PTR`.

Reply:
Okay, then using `ULONG_PTR` is better here.